config/logging.go

```diff
@@ -15,75 +15,72 @@ const (
-	PostLoggerLevel           string     `mapstructure:"post"`
-	StateDbLoggerLevel        string     `mapstructure:"stateDb"`
-	StateLoggerLevel          string     `mapstructure:"state"`
-	AtxHandlerLevel           string     `mapstructure:"atxHandler"`
-	AtxDbStoreLoggerLevel     string     `mapstructure:"atxDbStore"`
-	BeaconLoggerLevel         string     `mapstructure:"beacon"`
-	WeakCoinLoggerLevel       string     `mapstructure:"weakCoin"`
-	PoetDbStoreLoggerLevel    string     `mapstructure:"poetDbStore"`
-	StoreLoggerLevel          string     `mapstructure:"store"`
-	PoetDbLoggerLevel         string     `mapstructure:"poetDb"`
-	MeshDBLoggerLevel         string     `mapstructure:"meshDb"`
-	TrtlLoggerLevel           string     `mapstructure:"trtl"`
-	AtxDbLoggerLevel          string     `mapstructure:"atxDb"`
-	BlkEligibilityLoggerLevel string     `mapstructure:"block-eligibility"`
-	MeshLoggerLevel           string     `mapstructure:"mesh"`
-	SyncLoggerLevel           string     `mapstructure:"sync"`
-	FetcherLoggerLevel        string     `mapstructure:"fetcher"`
-	BlockOracleLevel          string     `mapstructure:"block-oracle"`
-	HareOracleLoggerLevel     string     `mapstructure:"hare-oracle"`
-	HareLoggerLevel           string     `mapstructure:"hare"`
-	BlockBuilderLoggerLevel   string     `mapstructure:"block-builder"`
-	BlockListenerLoggerLevel  string     `mapstructure:"block-listener"`
-	PoetListenerLoggerLevel   string     `mapstructure:"poet"`
-	NipostBuilderLoggerLevel  string     `mapstructure:"nipostBuilder"`
-	AtxBuilderLoggerLevel     string     `mapstructure:"atxBuilder"`
-	HareBeaconLoggerLevel     string     `mapstructure:"hare-beacon"`
-	TimeSyncLoggerLevel       string     `mapstructure:"timesync"`
-	VMLogLevel                string     `mapstructure:"vm"`
-	ProposalListenerLevel     string     `mapstructure:"proposalListener"`
+	Encoder LogEncoder `mapstructure:"log-encoder"`
+
+	AppLoggerLevel             string `mapstructure:"app"`
+	ClockLoggerLevel           string `mapstructure:"clock"`
+	P2PLoggerLevel             string `mapstructure:"p2p"`
+	PostLoggerLevel            string `mapstructure:"post"`
+	PostServiceLoggerLevel     string `mapstructure:"postService"`
+	StateDbLoggerLevel         string `mapstructure:"stateDb"`
+	BeaconLoggerLevel          string `mapstructure:"beacon"`
+	CachedDBLoggerLevel        string `mapstructure:"cachedDb"`
+	PoetDbLoggerLevel          string `mapstructure:"poetDb"`
+	TrtlLoggerLevel            string `mapstructure:"trtl"`
+	AtxHandlerLevel            string `mapstructure:"atxHandler"`
+	MeshLoggerLevel            string `mapstructure:"mesh"`
+	SyncLoggerLevel            string `mapstructure:"sync"`
+	HareOracleLoggerLevel      string `mapstructure:"hareOracle"`
+	BlockCertLoggerLevel       string `mapstructure:"blockCert"`
+	BlockGenLoggerLevel        string `mapstructure:"blockGenerator"`
+	BlockHandlerLoggerLevel    string `mapstructure:"blockHandler"`
```

Comment:
I feel like this needs to go into the `CHANGELOG.md`. It looks like originally it was named `block-listener` but has now been renamed to `blockHandler`, the same for `block-builder` / `blockGenerator`.

Reply:
The logging fields with invalid names didn't work at all. I explained in the PR description that the mapstructure tag must match the logger _name_. Check https://github.com/spacemeshos/go-spacemesh/blob/90fe8d5a080dc5f502d8d0b483f6b53baff3f615/node/node.go#L509-L515

There is nothing to be added to the changelog as this is not changing any functionality but fixing bugs :shrug: